debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -112,14 +113,24 @@ public BinlogReader(String name, MySqlTaskContext context) {
         EventDeserializer eventDeserializer = new EventDeserializer() {
             @Override
             public Event nextEvent(ByteArrayInputStream inputStream) throws IOException {
-                // Delegate to the superclass ...
-                Event event = super.nextEvent(inputStream);
-                // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
-                if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
-                    TableMapEventData tableMapEvent = event.getData();
-                    tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                try {
+                    // Delegate to the superclass ...
+                    Event event = super.nextEvent(inputStream);
+
+                    // We have to record the most recent TableMapEventData for each table number for our custom deserializers ...
+                    if (event.getHeader().getEventType() == EventType.TABLE_MAP) {
+                        TableMapEventData tableMapEvent = event.getData();
+                        tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
+                    }
+                    return event;
+                }
+                // DBZ-217 In case an event couldn't be read we create a pseudo-event for the sake of logging
+                catch(EventDataDeserializationException edde) {
+                    EventHeaderV4 header = new EventHeaderV4();
+                    header.setEventType(EventType.INCIDENT);
+                    EventData data = new EventDataDeserializationExceptionData(edde);
+                    return new Event(header, data);
```

Comment:
My concern is that `EventDataDeserializationException` is thrown anytime the binlog client library has an error reading the input stream, which will include situations where the binlog client library has a bug and is trying to read the event incorrectly. If that were the case, then this code would eat and log that event rather than failing.

The situation that we're trying to handle is a subset of cases where `EventDataDeserializationException` is thrown, and specifically the case where the binlog event is actually corrupted. Unfortunately, I don't see how we can discover that subset case, so I'm not sure we can do anything but what you've coded.

Bottom line is that I'm worried this change will make the connector eat/log more errors than it should, at which point those events are now lost and any data constructed using our event stream is potentially corrupt.